src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -120,44 +127,49 @@ public EventTypeService(final EventTypeRepository eventTypeRepository,
         }
     }
 
-    public Result<Void> delete(final String eventTypeName, final Client client) {
+    public void delete(final String eventTypeName, final Client client)
+            throws EventTypeDeletionException,
+            ForbiddenAccessException,
+            NoEventTypeException,
+            UnableProcessException {
         Closeable deletionCloser = null;
         try {
             deletionCloser = timelineSync.workWithEventType(eventTypeName, nakadiSettings.getTimelineWaitTimeoutMs());
 
             final Optional<EventType> eventTypeOpt = eventTypeRepository.findByNameO(eventTypeName);
             if (!eventTypeOpt.isPresent()) {
-                return Result.notFound("EventType \"" + eventTypeName + "\" does not exist.");
+                throw new NoEventTypeException("EventType \"" + eventTypeName + "\" does not exist.");
             }
             final EventType eventType = eventTypeOpt.get();
             if (!client.idMatches(eventType.getOwningApplication())) {
-                return Result.forbidden("You don't have access to this event type");
+                throw new ForbiddenAccessException("You don't have access to event type " + eventTypeName);
             }
             final List<Subscription> subscriptions = subscriptionRepository.listSubscriptions(
                     ImmutableSet.of(eventTypeName), Optional.empty(), 0, 1);
             if (!subscriptions.isEmpty()) {
-                return Result.conflict("Not possible to remove event-type as it has subscriptions");
+                throw new UnableProcessException("Can't remove event type " + eventTypeName
+                        + ", as it has subscriptions");
             }
-            final TopicRepository topicRepository = timelineService.getTopicRepository(eventType);
-            // TODO: Cascading delete. Event Type must be deleted with all it's timelines.
+            timelineService.deleteAllTimelinesForEventType(eventTypeName);
             eventTypeRepository.removeEventType(eventTypeName);
```

Comment:
I don't even know how to validate this place, if it is always possible to hack around: 
1. Create 10 timelines
2. Simultaneously delete event type and create subscription for it.
Timeline deletion takes time and probably after 10 attempts it will be possible to get event type that wasn't deleted because of constraints, but with all the topics deleted.